flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/slotpool/SlotPoolImplTest.java

```diff
@@ -738,6 +740,95 @@ public void testCalculationOfTaskExecutorUtilization() throws Exception {
 		}
 	}
 
+	@Test
+	public void testOrphanedAllocationCanBeRemapped() throws Exception {
+		try (SlotPoolImpl slotPool = createSlotPoolImpl()) {
+			final List<AllocationID> allocationIds = new ArrayList<>();
+			resourceManagerGateway.setRequestSlotConsumer(
+				slotRequest -> allocationIds.add(slotRequest.getAllocationId()));
+
+			final List<AllocationID> canceledAllocations = new ArrayList<>();
+			resourceManagerGateway.setCancelSlotConsumer(canceledAllocations::add);
+
+			setupSlotPool(slotPool, resourceManagerGateway, mainThreadExecutor);
+			final Scheduler scheduler = setupScheduler(slotPool, mainThreadExecutor);
+
+			final SlotRequestId slotRequestId1 = new SlotRequestId();
+			scheduler.allocateSlot(
+				slotRequestId1,
+				new DummyScheduledUnit(),
+				SlotProfile.noRequirements(),
+				timeout);
```

Comment:
```suggestion
			allocateSlot(scheduler, slotRequestId1);
```

Reply:
There is already a private method `SlotPoolImplTest#allocateSlot` for this.
I would also extend this private method to submit multiple allocations at once in order:
`allocateSlot(Scheduler scheduler, SlotRequestId ... slotRequestIds)`

Reply:
Done via introducing a `requestNewAllocatedSlots(SlotPool, SlotRequestId...)`.